lib/rpc/SingleJsonRpcProvider.ts

```diff
@@ -272,10 +273,12 @@ export class SingleJsonRpcProvider extends StaticJsonRpcProvider {
     metric.putMetric(`${this.metricPrefix}_selected`, 1, MetricLoggerUnit.Count)
   }
 
-  // Wrap another layer only for the sake of ease unit testing.
-  // We will test this API to represent the tests of other similar implemented APIs.
-  private _getBlockNumber(): Promise<number> {
-    return super.getBlockNumber()
+  logDbSync(success: boolean) {
```

Comment:
nit: since this is mostly syntactic sugar: thoughts on creating two function instead of one : 
```
logSuccessDbSync(){ metric.putMetric(`${this.metricPrefix}_db_sync_success`, 1, MetricLoggerUnit.Count) }
logFailureDbSync(){ metric.putMetric(`${this.metricPrefix}_db_sync_failure`, 1, MetricLoggerUnit.Count) }
```

This would save some navigation back and forth for the reader